src/search/engines/googleimages/googleimages.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/hearchco/hearchco/src/config"
+	"github.com/hearchco/hearchco/src/search/bucket"
+	"github.com/hearchco/hearchco/src/search/engines"
+	"github.com/hearchco/hearchco/src/search/engines/_sedefaults"
+	"github.com/hearchco/hearchco/src/search/result"
+	"github.com/rs/zerolog/log"
+)
+
+func Search(ctx context.Context, query string, relay *bucket.Relay, options engines.Options, settings config.Settings, timings config.Timings) error {
+	if err := _sedefaults.Prepare(Info.Name, &options, &settings, &Support, &Info, &ctx); err != nil {
+		return err
+	}
+
+	var col *colly.Collector
+	var pagesCol *colly.Collector
+	var retError error
+
+	_sedefaults.InitializeCollectors(&col, &pagesCol, &settings, &options, &timings)
+	// disable User Agent since Google Images responds with fake data if UA is correct
+	col.UserAgent = ""
+
+	_sedefaults.PagesColRequest(Info.Name, pagesCol, ctx)
+	_sedefaults.PagesColError(Info.Name, pagesCol)
+	_sedefaults.PagesColResponse(Info.Name, pagesCol, relay)
+
+	_sedefaults.ColRequest(Info.Name, col, ctx)
+	_sedefaults.ColError(Info.Name, col)
+
+	var pageRankCounter = make([]int, options.MaxPages*Info.ResultsPerPage)
+
+	col.OnResponse(func(e *colly.Response) {
+		body := string(e.Body)
+		index := strings.Index(body, "{\"ischj\":")
+
+		if index == -1 {
+			log.Error().
+				Str("body", body).
+				Msg("googleimages.Search() -> col.OnResponse: failed parsing response: failed finding start of JSON")
+			return
+		}
+
+		body = body[index:]
+		var jsonResponse JsonResponse
+		if err := json.Unmarshal([]byte(body), &jsonResponse); err != nil {
+			log.Error().
+				Str("body", body).
+				Msg("googleimages.Search() -> col.OnResponse: failed parsing response: failed unmarshalling JSON")
+			return
+		}
+
+		page := _sedefaults.PageFromContext(e.Request.Ctx, Info.Name)
+
+		for _, metadata := range jsonResponse.ISCHJ.Metadata {
+			origImg := metadata.OriginalImage
+			original := result.Image{
+				URL:    origImg.Url,
+				Height: origImg.Height,
+				Width:  origImg.Width,
+			}
+
+			thmbImg := metadata.Thumbnail
+			thumbnail := result.Image{
+				URL:    thmbImg.Url,
+				Height: thmbImg.Height,
+				Width:  thmbImg.Width,
+			}
+
+			resultJson := metadata.Result
+			textInGridJson := metadata.TextInGrid
+
+			if resultJson.ReferrerUrl != "" && original.URL != "" && thumbnail.URL != "" {
+				res := bucket.MakeSEImageResult(resultJson.ReferrerUrl, resultJson.PageTitle, textInGridJson.Snippet, resultJson.SiteTitle, original, thumbnail, Info.Name, page, pageRankCounter[page]+1)
+				bucket.AddSEResult(res, Info.Name, relay, &options, pagesCol)
+				pageRankCounter[page]++
+			} else {
+				log.Error().
+					Str("engine", Info.Name.String()).
+					Str("jsonMetadata", fmt.Sprintf("%v", metadata)).
+					Str("url", resultJson.ReferrerUrl).
+					Str("original", original.URL).
+					Str("thumbnail", thumbnail.URL).
+					Msg("googleimages.Search() -> onHTML: Couldn't find image URL")
```

Comment:
Another PR that will remove functions names from logs will be made, since zerolog already prints file and line of code which produces the log in `-vv` (trace) mode.